pkg/controller/operators/catalog/operator.go

```diff
@@ -997,6 +997,7 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
+			o.resolver.DumpCache(namespace)
```

Comment:
While I can guess the reason behind exposing more information regarding what operators are available in the namespace for diagnostic reason, the fact there are lot of operators in default catalogsources which are available for all namespaces, this print-out can potentially overwhelm the logs for catalog-operator which is already crowded. You may want to consider if this is really a `Infof` or `Debugf` depending what the targeted audience this info is for. Admittedly, if this is debug log, it will not be present in log unless it is enabled manually which can be a pain to ask from the customer.

Secondly, this particular condition check will return a nil AKA this is not a terminal condition. So this will lead to retry on resolver. Continuously printing this info over and over again on the log is certainly not a good idea.

Lastly, I don't see how this dump will help with identifying with where this issue comes from in term of "is this catalog's fault AKA missing satisfiable candidates or is this existing operator's fault AKA conflicting stuff". It is important for logging to be helpful and not just for sake of getting the haystack first and hopefully find the needle later because it is a difficult task either way.